### PR TITLE
Magnum Opus: Slot Machine

### PR DESCRIPTION
**Pull request by NoahTheDuke**
Done did.

### CODE REVIEW

File: src/clj/game/cards/ice.clj [hunk 1]
```diff
@@ -2104,6 +2104,38 @@
...
+    :abilities [{:label "Roll them bones"
+                 :effect (effect (move :runner (first (:deck runner)) :deck)
+                                 (system-msg (str "uses Slot Machine to put the top card of the stack to the bottom,"
+                                                  " then reveal the top 3 cards in the stack: "
+                                                  (join ", " (map :title (take 3 (:deck runner)))))))}]
+    :subroutines [{:label "Runner loses 3 [credits]"
+                   :msg "Runner loses 3 [credits]"
+                   :effect (effect (lose-credits :runner 3))}
+                  {:label "Corp gains 3 [credits]"
+                   :effect (req (let [f (:type (nth (:deck runner) 0))
```
Comment (Saintis) [inline src/clj/game/cards/ice.clj]: I feel there is a nicer way of doing this involving list expansion and `take`.
E.g. `[unique-types (count (into #{} (map :type (take 3 (:deck runner)))))]`
Or as a threading macro: `[unique-types (-> (:deck runner) (take 3) (map :type) (into #{}) count)]`

And then checking if unique-subtypes <= 2 and == 1.

File: src/clj/game/cards/ice.clj [hunk 2]
```diff
@@ -2104,6 +2104,38 @@
                    :async true
                    :effect (req (do-access state :runner eid [:rd] {:no-root true}))}]}
 
+   "Slot Machine"
+   {:implementation "Encounter effect is manual"
+    :abilities [{:label "Roll them bones"
```
Comment (Saintis) [inline src/clj/game/cards/ice.clj]: This is perfect as it.

File: src/clj/game/cards/ice.clj [hunk 3]
```diff
@@ -2104,6 +2104,38 @@
...
+                 :effect (effect (move :runner (first (:deck runner)) :deck)
+                                 (system-msg (str "uses Slot Machine to put the top card of the stack to the bottom,"
+                                                  " then reveal the top 3 cards in the stack: "
+                                                  (join ", " (map :title (take 3 (:deck runner)))))))}]
```
Comment (Saintis) [inline src/clj/game/cards/ice.clj]: I wonder if we want to help the mental math and print the types of each card here as well. E.g. "Sure Gamble (event), Career Fair (event), Bankroll (program)".

File: src/clj/game/cards/ice.clj [hunk 4]
```diff
@@ -2104,6 +2104,38 @@
...
+    :subroutines [{:label "Runner loses 3 [credits]"
+                   :msg "Runner loses 3 [credits]"
+                   :effect (effect (lose-credits :runner 3))}
+                  {:label "Corp gains 3 [credits]"
```
Comment (Saintis) [inline src/clj/game/cards/ice.clj]: This should just be Gain 3 credits I think
```suggestion
                  {:label "Gain 3 [credits]"
```

File: src/clj/game/cards/ice.clj [hunk 5]
```diff
@@ -2104,6 +2104,38 @@
...
+                   :effect (req (let [f (:type (nth (:deck runner) 0))
+                                      s (:type (nth (:deck runner) 1))
+                                      t (:type (nth (:deck runner) 2))]
+                                 (when (or (= f s)
+                                           (= f t)
+                                           (= s t))
+                                   (system-msg state :corp (str "gains 3 [credits]"))
+                                   (gain-credits state :corp 3))))}
+                  {:label "Corp places 3 advancement tokens"
```
Comment (Saintis) [inline src/clj/game/cards/ice.clj]: Similar to other label comment
```suggestion
                  {:label "Place 3 advancement tokens"
```

File: src/clj/game/cards/ice.clj (hunk 3 above)
Comment (NoahTheDuke) [reply src/clj/game/cards/ice.clj]: Very true! I'll do that.

File: src/clj/game/cards/ice.clj (hunk 1 above)
Comment (NoahTheDuke) [reply src/clj/game/cards/ice.clj]: Oh that's kind of cool. I like that. I'll add it in!

**Pull request merged**